File: Missing-Number.java
public class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int total = n * (n + 1) / 2;
        int actual = 0;
        for (int num : nums) {
            actual += num;
        }
        return total - actual;
    }
}

File: Missing-Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n = nums.length;
    let total = n * (n + 1) / 2;
    let actual = nums.reduce((a, b) => a + b, 0);
    return total - actual;
};
